task/jefe/src/main.rs

```diff
@@ -252,6 +253,20 @@ impl idol_runtime::NotificationHandler for ServerImpl<'_> {
                 }
             }
         }
+        // Check whether all 'exit' tasks have 'Exited'
+        let mut done: bool = true;
+        for (i, exits) in TASK_EXITS.iter().enumerate() {
+            if *exits
+                && (kipc::read_task_status(i)
+                    != abi::TaskState::Healthy(abi::SchedState::Exited))
+            {
+                done = false;
+                break;
+            }
+        }
+        if done {
+            kipc::system_restart();
```

Comment:
We could probably make this a config in the app.toml for now.  I think the current implementation could cause problems for apps that have a one shot task at startup, then continue running "forever" with some driver services.

Reply:
It would be a set of features for the `jefe` task and the appropriate feature can be selected by the app as per its requirements in app.toml/Cargo.toml. Is that what you meant by config in app.toml?

Reply:
Yea, I was thinking more with the first 3 cases you listed.  I am not sure if that provides a good way of integrating #4.

Reply:
I think the same logic should work for #4 as well. Instead of making `kipc::system_restart` or `kipc::poweroff` call, `jefe` would make a call like `kipc::exit` or `kipc::jump_to` or some better named kipc.

Reply:
"system-restart-when-done" feature for jefe added to select this behavior.